### PR TITLE
Fix for pagination per page

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

File: custom-recipes/api-connect/recipe.json [hunk 1]
```diff
@@ -169,16 +169,21 @@
...
                 {"value": "next_page", "label": "Next page URL provided"},
                 {"value": "offset", "label": "Offset pagination"},
-                {"value": "page", "label": "Page pagination"}
+                {"value": "page", "label": "Per page"}
             ]
         },
+        {
+            "type": "SEPARATOR",
+            "description": "&#9888;'Per page' mode requires the 'Key to data array' to be defined",
+            "visibilityCondition": "model.pagination_type=='page'"
```
Comment (HenriChabert) [inline custom-recipes/api-connect/recipe.json:185]: Could you check that "Key to data array" is already filled and if so, remove the warning message ?
```suggestion
            "visibilityCondition": "model.pagination_type=='page' && model.extraction_key"
```

File: custom-recipes/api-connect/recipe.json [hunk 2]
```diff
@@ -193,7 +198,7 @@
             "description": "",
             "type": "STRING",
             "defaultValue": null,
-            "visibilityCondition": "(['page', 'offset'].indexOf(model.pagination_type)>-1)"
+            "visibilityCondition": "(['offset'].indexOf(model.pagination_type)>-1)"
```
Comment (HenriChabert) [inline custom-recipes/api-connect/recipe.json:201]: Maybe change it as this parameter is now only her for what pagination_type
```suggestion
            "visibilityCondition": "model.pagination_type == 'offset'"
```

File: custom-recipes/api-connect/recipe.json [hunk 3]
```diff
@@ -193,7 +198,7 @@
...
         },
         {
             "name": "skip_key",
```
Comment (HenriChabert) [inline custom-recipes/api-connect/recipe.json:204]: This parameter seems te same than the next one. Can't it be gathered in a unique parameter? With the following visibility condition:
`"visibilityCondition": "(['page', 'offset'].indexOf(model.pagination_type)>-1)"`

File: python-connectors/api-connect_dataset/connector.json [hunk 4]
```diff
@@ -129,16 +129,21 @@
...
                 {"value": "offset", "label": "Offset pagination"},
-                {"value": "page", "label": "Page pagination"}
+                {"value": "page", "label": "Per page"}
             ]
         },
+        {
+            "type": "SEPARATOR",
+            "description": "&#9888; Requires a key to data array",
+            "visibilityCondition": "model.pagination_type=='page'"
+        },
```
Comment (HenriChabert) [inline python-connectors/api-connect_dataset/connector.json:146]: Same remarks than the recipe

File: python-lib/pagination.py [hunk 5]
```diff
@@ -49,6 +52,7 @@ def set_counting_key(self, counting_key):
 
     def update_next_page(self, data):
         self.is_first_batch = False
+        self.counter = self.counter + 1
```
Comment (HenriChabert) [inline python-lib/pagination.py:55]: Nitpick: `self.counter += 1`

**Review by HenriChabert — APPROVED** [review]
LGTM!
Some very minor remarks.
about the pagination, I would have personnally used inheritance with a parent pagination engine and then for each pagination type, a small class with the specific behaviour. But it is probably overkill in this case!

File: custom-recipes/api-connect/recipe.json (hunk 1 above)
Comment (alexbourret) [reply custom-recipes/api-connect/recipe.json:185]: 👍

**Pull request merged**